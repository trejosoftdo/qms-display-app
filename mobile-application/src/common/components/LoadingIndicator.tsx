import * as React from 'react';
import { StyleSheet } from 'react-native';
import { ActivityIndicator } from 'react-native-paper';
import AppLightTheme from '../theme';


/**
 * LoadingIndicatorProps defines the props for the Loading Indicator Component.
 *
 * @property loading - to indicate it is loading
 */
interface LoadingIndicatorProps {
  loading?: boolean;
}


/**
 * A component that indicates the app is busy
 *
 * @param {LoadingIndicatorProps} props - The props for the Loading Indicator component.
 */
const LoadingIndicator: React.FC<LoadingIndicatorProps> = (props: LoadingIndicatorProps) => (
  <ActivityIndicator
    style={styles.container}
    animating={props.loading}
    size="large"
    color={AppLightTheme.colors.primary}
  />
);

const styles = StyleSheet.create({
  container: {
    marginTop: 10,
    marginBottom: 10,
  },
});

export default LoadingIndicator;
