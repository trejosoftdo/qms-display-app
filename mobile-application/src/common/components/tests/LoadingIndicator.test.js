import React from 'react';
import LoadingIndicator from '../LoadingIndicator';


describe('LoadingIndicator component', () => {
  it('renders the component with the provided props as expected', () => {
    expect(renderShallow(<LoadingIndicator loading />)).toMatchSnapshot();
  });
});