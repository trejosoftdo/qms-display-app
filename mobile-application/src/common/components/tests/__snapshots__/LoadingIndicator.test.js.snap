// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator component renders the component with the provided props as expected 1`] = `
<ActivityIndicator
  animating={true}
  color="#2929cc"
  size="large"
  style={
    {
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
/>
`;
